### ISSUE TITLE
 Change state from Call for Records  when a Fee has bee paid to Deduplication or Records Review

### ISSUE THREAD

**Issue by lmullane** [bcgov/foi-flow#2201]
* As a (type of user)
* I want (goal) 
* so that (reason/ why/ value) 

**Assumptions & Scope**	
What are the assumptions for this story?

What is IN scope?

What is NOT in scope?

**Acceptance Criteria**	

**Scenario 1: Change state from Call for Records  when a Fee has bee paid to Deduplication or Records Review**
* GIVEN a ministry coordinator has completed the actual hours and saved the CFR Form view 
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/912622803
* AND a fee has been levied and paid by the applicant
* WHEN the coordinator clicks on Deduplication or Records Review in the state menu dropdown
* THEN the state model will open
* AND uploading a CFR form will no longer be required for Records Review
* AND the blue 'Save Changes' button will be active

**Scenario 2: Fail to change state from Call for Records to when a Fee has been paid to Deduplication or Records Review*
* GIVEN a ministry coordinator has NOT completed the actual hours and saved the CFR Form view 
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/912622803
* AND a fee has been levied and paid by the applicant
* WHEN the coordinator clicks on Deduplication or Records Review in the state menu dropdown
* THEN the state model will open
* AND an error message will be displayed: "To update the state you must first complete the actual hours in the CFR Form (insert link)"
* AND the blue 'Save Changes' button will be inactive

*Scenario 3: Change state from Call for Records to when a fee has not been paid Deduplication or Records Review**
* GIVEN a ministry coordinator has completed the estimated hours and saved the CFR Form view and the Total Fees Due is zero
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/912622803
* AND a fee has NOT been levied and paid by the applicant
* WHEN the coordinator clicks on Deduplication or Records Review in the state menu dropdown
* THEN the state model will open
* AND uploading a CFR form will no longer be required for Records Review
* AND the blue 'Save Changes' button will be active

**Scenario 4: Fail to change state from Call for Records to Deduplication or Records Review*
* GIVEN a ministry coordinator has NOT completed the estimated hours and saved the CFR Form view 
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/912622803
* AND a fee has been NOT been levied and paid by the applicant
* WHEN the coordinator clicks on Deduplication or Records Review in the state menu dropdown
* THEN the state model will open
* AND an error message will be displayed: "To update the state you must first complete the actual hours in the CFR Form (insert link)"
* AND the blue 'Save Changes' button will be inactive

...

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by lmullane** [comment]
This story has a dependency with applicant having paid (i.e., those stories must be complete first).

**Comment by lmullane** [comment]
Looks like this may be a duplicate now of #2198
